lib/overcommit/hook/pre_push/r_spec.rb

```diff
@@ -1,14 +1,47 @@
 module Overcommit::Hook::PrePush
   # Runs `rspec` test suite before push
+  # .overcommit.yml configuration:
+  # Set `changed: true` to run rspec only over edited files.
+  # Set `remote`: $remote_name` to specify a remote name.
   #
   # @see http://rspec.info/
   class RSpec < Base
     def run
-      result = execute(command)
+      command = ['rspec']
+      remote = config['remote'] || 'origin'
+      result = if config['changed']
+                 return :pass if testable_files(remote).empty?
+                 execute([*command, *testable_files(remote)])
```

Comment:
`testable_files` is called twice here; we should memoize the first call so we aren't recalculating unnecessarily.